pkg/proxy/config/api_test.go

```diff
@@ -174,24 +165,16 @@ func TestNewEndpointsSourceApi_UpdatesAndMultipleEndpoints(t *testing.T) {
 
 	ch := make(chan EndpointsUpdate)
 
-	cache.NewReflector(lw, &api.Endpoints{}, NewEndpointsStore(nil, ch), 30*time.Second).Run()
-
-	got, ok := <-ch
-	if !ok {
-		t.Errorf("Unable to read from channel when expected")
-	}
-	expected := EndpointsUpdate{Op: SET, Endpoints: []api.Endpoints{}}
-	if !apiequality.Semantic.DeepEqual(expected, got) {
-		t.Errorf("Expected %#v; Got %#v", expected, got)
-	}
+	endpointsController := NewEndpointsController(lw, 30*time.Second, ch)
+	go endpointsController.Run(wait.NeverStop)
```

Comment:
So regarding one - it seems it's not easy to handle (and this is not the regression).
Basically, both before and now, with the initial "LIST" operation (of the underlying reflector), we are sending the ADD events from the results of it one by one, and then passing to proxier.

We need a better way to handle it, but since this is not a regression, I would prefer fixing it in a separate PR (I can try sending it later today).